src/System.Linq/src/System/Linq/Partition.cs

```diff
@@ -163,7 +163,7 @@ IEnumerator IEnumerable.GetEnumerator()
         public IPartition<TElement> Skip(int count)
         {
             int minIndex = _minIndex + count;
-            return minIndex >= _maxIndex ? EmptyPartition<TElement>.Instance : new OrderedPartition<TElement>(_source, minIndex, _maxIndex);
+            return (uint)minIndex > (uint)_maxIndex ? EmptyPartition<TElement>.Instance : new OrderedPartition<TElement>(_source, minIndex, _maxIndex);
```

Comment:
Replacing `>=` with `>` is the fix for the off-by-one. Casting to `(uint)` is the fix for the overflow behaviour.